src/fides/api/ops/api/v1/endpoints/user_permission_endpoints.py

```diff
@@ -94,11 +95,22 @@ def update_user_permissions(
         user.client.update(
             db=db, data={"scopes": permissions.scopes, "roles": permissions.roles}
         )
-    return user.permissions.update(  # type: ignore[attr-defined]
+    updated_user_perms = user.permissions.update(  # type: ignore[attr-defined]
         db=db,
         data={"id": user.permissions.id, "user_id": user_id, "scopes": permissions.scopes, "roles": permissions.roles},  # type: ignore[attr-defined]
     )
 
+    if user.systems and APPROVER in user.permissions.roles:  # type: ignore[attr-defined]
+        for system in user.systems.copy():
+            logger.info(
+                "Approvers cannot be system managers. Removing user {} as system manager of {}.",
+                user.id,
+                system.fides_key,
+            )
+            user.remove_as_system_manager(db, system)
+
+    return updated_user_perms
```

Comment:
Not specified, but removing users as system managers if they are being made an Approver. This happens behind the scenes. Should we add a confirmation or alert that it's happening if we go this route?

Reply:
Yes, I think that's reasonable. Can we file a ticket for this?